cli/genpolicy/genpolicy.go

```diff
@@ -0,0 +1,124 @@
+	aksCloudHypervisorSNPRules []byte
+	//go:embed assets/allow-all.rego
+	permissiveRules []byte
+)
+
+// Runner is a wrapper around the genpolicy tool.
+//
+// Create an instance with New(), optionally configure it using the exported
+// fields, call Run() to execute the tool, and call Teardown() afterwards to
+// clean up temporary files.
+type Runner struct {
+	Rules     []byte
+	Settings  []byte
+	CacheFile string
+
+	once      sync.Once
+	genpolicy embedbin.Installed
+	settings  embedbin.Installed
+	rules     embedbin.Installed
+}
+
+// New creates a new Runner with the default configuration for the requested platform.
+func New(platform platforms.Platform) (*Runner, error) {
+	cacheDir, err := os.UserCacheDir()
+	if err != nil {
+		return nil, fmt.Errorf("getting cache dir: %w", err)
+	}
+	runner := &Runner{
+		Settings:  defaultGenpolicySettings,
+		CacheFile: filepath.Join(cacheDir, "contrast", "layer-cache.json"),
+	}
+	switch platform {
+	case platforms.AKSCloudHypervisorSNP:
+		runner.Rules = aksCloudHypervisorSNPRules
+	default:
+		// TODO(burgerdev): use real rules for supported platforms.
+		runner.Rules = permissiveRules
+	}
+
+	return runner, nil
+}
+
+// Run runs the tool on the given yaml.
+//
+// Run can be called more than once, but if the first run fails during
+// initialization subsequent runs won't succeed.
+func (r *Runner) Run(ctx context.Context, yamlPath string, logger *slog.Logger) error {
+	var outerErr error
+	r.once.Do(func() {
+		var err error
+		e := embedbin.New()
+		r.genpolicy, err = e.Install("", genpolicyBin)
+		outerErr = errors.Join(outerErr, err)
+		r.settings, err = e.Install("", r.Settings)
+		outerErr = errors.Join(outerErr, err)
+		r.rules, err = e.Install("", r.Rules)
+		outerErr = errors.Join(outerErr, err)
+	})
+	if outerErr != nil {
+		return fmt.Errorf("creating temp files: %w", outerErr)
+	}
+
+	if err := os.MkdirAll(filepath.Dir(r.CacheFile), os.ModePerm); err != nil {
+		return fmt.Errorf("creating cache file: %w", err)
+	}
+
+	args := []string{
+		"--runtime-class-names=contrast-cc",
+		"--rego-rules-path=" + r.rules.Path(),
+		"--json-settings-path=" + r.settings.Path(),
+		"--yaml-file=" + yamlPath,
+		"--layers-cache-file-path=" + r.CacheFile,
+	}
+	genpolicy := exec.CommandContext(ctx, r.genpolicy.Path(), args...)
+	genpolicy.Env = append(genpolicy.Env, "RUST_LOG=info", "RUST_BACKTRACE=1")
+	genpolicy.Env = append(genpolicy.Env, fmt.Sprintf("PATH=%s", os.Getenv("PATH")))
+
+	logFilter := newLogTranslator(logger)
+	defer logFilter.stop()
+	genpolicy.Stdout = io.Discard
+	genpolicy.Stderr = logFilter
+
+	if err := genpolicy.Run(); err != nil {
+		return fmt.Errorf("running genpolicy: %w", err)
+	}
+
+	return nil
+}
+
+// Teardown cleans up temporary files and should be called after the last Run.
+func (r *Runner) Teardown() error {
+	var err error
+	for _, install := range []embedbin.Installed{r.genpolicy, r.settings, r.rules} {
+		err = errors.Join(err, install.Uninstall())
+	}
```

Comment:
This will cause a null-pointer exception if `Runner.Run` has never been called and `genpolicy` etc haven't been initialized.

Reply:
I'm checking for nil now.